src/go/k8s/internal/controller/redpanda/redpanda_controller_test.go

```diff
@@ -0,0 +1,234 @@
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	logger := testr.New(t).V(0)
+	log.SetLogger(logger)
+	ctx = log.IntoContext(ctx, logger)
+
+	var cluster *k3d.Cluster
+	var err error
+	if val, ok := os.LookupEnv("EXISTING_K3D_CLUSTER"); ok && val != "" {
+		cluster, err = k3d.ExistingCluster(val)
+		require.NoError(t, err)
+		t.Logf("imported cluster %T %q", cluster, cluster.Name)
+	} else {
+		cluster, err = k3d.NewCluster(t.Name())
+		require.NoError(t, err)
+		t.Logf("created cluster %T %q", cluster, cluster.Name)
+	}
+
+	t.Cleanup(func() {
+		if testutil.Retain() {
+			t.Logf("retain flag is set; not deleting cluster %q", cluster.Name)
+			return
+		}
+		t.Logf("Deleting cluster %q", cluster.Name)
+		require.NoError(t, cluster.Cleanup())
+	})
+
+	h := helmtest.Setup(t)
+
+	n := h.Namespaced(t)
+
+	require.NoError(t, redpandav1alpha2.AddToScheme(scheme.Scheme))
+	require.NoError(t, cmapiv1.AddToScheme(scheme.Scheme))
+
+	c, err := client.New(cluster.RESTConfig(), client.Options{})
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		if testutil.Retain() {
+			t.Logf("retain flag is set; not deleting namespace %q", n.Namespace())
+			return
+		}
+		t.Logf("Deleting namespace %q", n.Namespace())
+		require.NoError(t, c.Delete(context.Background(), &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{Name: n.Namespace()},
+		}))
+	})
+
+	opt := envtest.CRDInstallOptions{
+		Scheme: scheme.Scheme,
+		Paths:  []string{filepath.Join("../", "../", "../", "config", "crd", "bases")},
+	}
+
+	_, err = envtest.InstallCRDs(cluster.RESTConfig(), opt)
+	require.NoError(t, err)
+
+	rpName := "smoke-test"
+
+	rp := &redpandav1alpha2.Redpanda{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      rpName,
+			Namespace: n.Namespace(),
+		},
+		Spec: redpandav1alpha2.RedpandaSpec{
+			ChartRef: redpandav1alpha2.ChartRef{},
+			ClusterSpec: &redpandav1alpha2.RedpandaClusterSpec{
+				Image:       &redpandav1alpha2.RedpandaImage{Tag: ptr.To("v24.2.4")},
+				Console:     &redpandav1alpha2.RedpandaConsole{Enabled: ptr.To(false)},
+				Statefulset: &redpandav1alpha2.Statefulset{Replicas: ptr.To(1)},
+			},
+		},
+	}
+	require.NoError(t, c.Create(ctx, rp))
+
+	t.Cleanup(func() {
+		t.Log("Deleting Redpanda CR smoke-test")
+		err = c.Delete(context.Background(), &redpandav1alpha2.Redpanda{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      rpName,
+				Namespace: n.Namespace(),
+			},
+		})
+
+		require.NoError(t, client.IgnoreNotFound(err))
+	})
+
+	// Start up our manager
+	mgr, err := manager.New(cluster.RESTConfig(), manager.Options{
+		Metrics: metricsserver.Options{BindAddress: "0"},
+		BaseContext: func() context.Context {
+			return log.IntoContext(ctx, logger)
+		},
+	})
+	require.NoError(t, err)
+
+	r := RedpandaReconciler{Client: c}
+	require.NoError(t, r.SetupWithManager(ctx, mgr))
+
+	testutils.TGo(t, ctx, func(ctx context.Context) error {
+		return mgr.Start(log.IntoContext(ctx, logger))
```

Comment:
hasn't `log.IntoContext` already been called?

Reply:
You are right, it can be removed.